### DISCUSSION TITLE
Main difference between using decode() and transcribe() methods

### DISCUSSION THREAD

**Discussion by kevinco26** [openai/whisper#1286] [Q&A] [answered]
Hi!
I've been doing some tests on both functions and can't seem to understand the difference.

I'm trying to get speech to text of an audio file.
One way i've done it is:
```
model = whisper.load_model("base")
mel = whisper.log_mel_spectrogram('audio'.mp3).to(model.device)
options = whisper.DecodingOptions(fp16=False)
result = whisper.decode(model, mel, options)
```

Output is fine.

Then I also tried
`result = model.transcribe('audio'.mp3, fp16=False)`

Output is the same and time is the very similar

Any suggestions/ideas/explanations for the two?

**Comment by localundine** [comment, marked as answer]
Hey, from what I can tell looking at the source code, transcribe is a wrapper around some common transcription methods, including decode. The transcribe method is basically there so you can do things like decode and load model in one simple line. Transcribe actually calls decode internally, so in the sample code you put up you're essentially doing pretty much the same thing, except in one case you're using a method designed to make things simpler and in the other case you're somewhat rebuilding the transcribe method.
